libgnucash/engine/qof-string-cache.cpp

```diff
@@ -43,98 +41,60 @@ static QofLogModule log_module = QOF_MOD_UTIL;
 /* =================================================================== */
 /* The QOF string cache                                                */
 /*                                                                     */
-/* The cache is a GHashTable where a copy of the string is the key,    */
+/* The cache is an std:unordered_map where an std::string is the key,  */
 /* and a ref count is the value                                        */
 /* =================================================================== */
 
-static GHashTable* qof_string_cache = NULL;
-
-static GHashTable*
-qof_get_string_cache(void)
-{
-    if (!qof_string_cache)
-    {
-        qof_string_cache = g_hash_table_new_full(
-                               g_str_hash,               /* hash_func          */
-                               g_str_equal,              /* key_equal_func     */
-                               g_free,                   /* key_destroy_func   */
-                               g_free);                  /* value_destroy_func */
-    }
-    return qof_string_cache;
-}
+static std::unordered_map<std::string, int> cache;
 
 void
 qof_string_cache_init(void)
 {
-    (void)qof_get_string_cache();
+    // nop
 }
 
 void
 qof_string_cache_destroy (void)
 {
-    if (qof_string_cache)
-    {
-        g_hash_table_destroy(qof_string_cache);
-    }
-    qof_string_cache = NULL;
+    cache.clear ();
```

Comment:
Unnecessary, and since `qof_string_cache_init` and `qof_string_cache_destroy` are each used only twice--once in the unit test and once in qofutil.cpp--it's better to just remove them.